### PR TITLE
3.3 plugin features

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/classes/Elgg/Composer/PostInstall.php [hunk 1]
```diff
@@ -94,6 +61,6 @@ private static function symlinkPluginFromRootToElgg($plugin) {
 		$from = Directory\Local::projectRoot()->getPath("mod/$plugin");
 		$to = Elgg\Application::elggDir()->getPath("mod/$plugin");
 
-		return !file_exists($from) && symlink($to, $from);
+		return !is_dir($from) && symlink($to, $from);
```
Comment (jeabakker) [inline engine/classes/Elgg/Composer/PostInstall.php]: wrong advise from me, please revert this

File: engine/classes/ElggPlugin.php [hunk 2]
```diff
@@ -848,6 +848,10 @@ public function register() {
 	public function boot() {
 		$result = null;
 		if ($this->canReadFile('start.php')) {
+			if (!in_array($this->getID(), \Elgg\Database\Plugins::BUNDLED_PLUGINS)) {
+				elgg_deprecated_notice("Using a start file in your plugin [{$this->getID()}] is deprecated. Use a elgg-plugin.php or PluginBootstrap class for your plugin.", '3.3');
```
Comment (jeabakker) [inline engine/classes/ElggPlugin.php]: using a start.php ....

**Review by jeabakker — CHANGES_REQUESTED** [review]
no tests yet??

**Comment by jdalsem** [comment]
Created https://github.com/Elgg/Elgg/issues/13014 to add more tests for elgg-plugin.php features

**Review by jeabakker — APPROVED** [review]

**Pull request merged**